lib/livebook/session.ex

```diff
@@ -510,10 +512,11 @@ defmodule Livebook.Session do
 
   defp schedule_autosave(state) do
     if interval_s = state.data.notebook.autosave_interval_s do
-      ref = Process.send_after(self(), :autosave, interval_s * 1000)
-      %{state | autosave_timer_ref: ref}
+      message_ref = make_ref()
+      ref = Process.send_after(self(), {:autosave, message_ref}, interval_s * 1000)
```

Comment:
You don't need to have a unique ref alongside the timer. What you can do is something like this instead:

```
if Process.cancel_timer(ref) == false do
  receive do
    :autosave -> true
  end
end
```

Reply:
Nice!